lib/app/shared/design_system/widgets/buttons/styles/filled_button_style.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/material.dart';
+
+import '../../../styles/penhas_colors.dart';
+import '../../../styles/penhas_text_style.dart';
+import 'penhas_button_style.dart';
+
+/// Uma implementação personalizada do [PenhasButtonStyle] para botão plano preenchido.
+///
+/// Este estilo de botão define uma série de propriedades para personalizar
+/// a aparência e o comportamento de um botão plano dentro da aplicação.
+///
+/// Usa cores e textos de [PenhasColors] e [PenhasTextStyle] para manter
+/// a consistência visual.
+///
+/// Exemplo de uso:
+/// ```dart
+/// PenhasButton(
+///   onPressed: () {},
+///   style: FilledButtonStyle(),
+///   child: Text('Botão Plano'),
+/// )
+/// ```
+///
+/// Veja também:
+/// * [PenhasButtonStyle], a interface que esta classe implementa.
+/// * [PenhasColors], onde as cores usadas são definidas.
+/// * [PenhasTextStyle], onde os estilos de texto usados são definidos.
+class FilledButtonStyle implements PenhasButtonStyle {
```

Comment:
Seria possível extender `ButtonStyle`?

Reply:
Na verdade, o que acredito que seria bem interessante é se de alguma maneira pudesse extender o tema atual para que não seja necessário definir algumas coisas novamente, como o `fontFamily`.

Reply:
> Na verdade, o que acredito que seria bem interessante é se de alguma maneira pudesse extender o tema atual para que não seja necessário definir algumas coisas novamente, como o `fontFamily`.

O `ButtonStyle` não herda esta informação, ele é instanciado com estas informações da hierarquia do Button. A única alternativa é utilizar pegar o `ButtonStyle` de uma API de botão do sistema e aplicar a atualização.

Então as alternativas que temos aqui são:

# 1

Mantenha o `PenhasButtonStyle` e na implementação faz um override com os valores desejados. Como o indicado no [Restoring the original button visuals](https://docs.flutter.dev/release/breaking-changes/buttons#restoring-the-original-button-visuals).

```dart
class RoundedOutlinedButtonStyle implements PenhasButtonStyle {
  @override
  ButtonStyle get buttonStyle => OutlinedButton.styleFrom(
        primary: PenhasColors.white,
        backgroundColor: Colors.transparent,
        shape: const RoundedRectangleBorder(
          borderRadius: BorderRadius.all(Radius.circular(20.0)),
          side: BorderSide(color: PenhasColors.white),
        ),
        textStyle: PenhasTextStyle.labelLarge.copyWith(
          color: PenhasColors.white,
        ),
      );
}
```

# 2 

Implementar a partir do `ButtonStyle` nos obriga a fazer um override de mais de 20 properties

```dart
class RoundedOutlinedButtonStyle implements ButtonStyle {
  @override
  AlignmentGeometry? get alignment => OutlinedButton.styleFrom().alignment;

  @override
  Duration? get animationDuration => OutlinedButton.styleFrom().animationDuration;

  @override
  MaterialStateProperty<Color?>? get backgroundColor => MaterialStateProperty.all<Color>(Colors.transparent);

  ...
 }
```

# 3

Cria o widget de button e aplica o estilo diretamente. Esta abordagem contribui na leitura do código, pois está objetivo qual widget e visual que será utilizado.

```dart
class PenhasRoundedOutlinedButton extends OutlinedButton {
  PenhasRoundedOutlinedButton({
    Key? key,
    required VoidCallback? onPressed,
    required Widget child,
  }) : super(
          key: key,
          onPressed: onPressed,
          child: child,
          style: OutlinedButton.styleFrom(
            primary: PenhasColors.white,
            backgroundColor: Colors.transparent,
            shape: const RoundedRectangleBorder(
              borderRadius: BorderRadius.all(Radius.circular(20.0)),
              side: BorderSide(color: PenhasColors.white),
            ),
            textStyle: PenhasTextStyle.labelLarge.copyWith(
              color: PenhasColors.white,
            ),
          ),
        );
}
```

Reply:
Concordo contigo com relação a terceira abordagem, fica bem mais fácil a leitura, mas pelo modo que hoje o código esteja estruturado, o ideal seria a primeira.
O que acha de aplicar ela e seguir a estrutura feita no `PenhasTextStyle`?

```dart
class PenhasButtonStyle {
    static ButtonStyle get roundedFilled  => // ...

    static ButtonStyle get roundedOutlined  => // ...
}
``` 

Aí poderia ficar localizado no package de styles.

Reply:
> Concordo contigo com relação a terceira abordagem, fica bem mais fácil a leitura, mas pelo modo que hoje o código esteja estruturado, o ideal seria a primeira. O que acha de aplicar ela e seguir a estrutura feita no `PenhasTextStyle`?
> 
> ```dart
> class PenhasButtonStyle {
>     static ButtonStyle get roundedFilled  => // ...
> 
>     static ButtonStyle get roundedOutlined  => // ...
> }
> ```
> 
> Aí poderia ficar localizado no package de styles.

Esta abordagem fere o **Single-responsibility principle**. Toda vez que for necessário criar um novo estilo, será necessário utilizar uma de duas alternativas: 

1. Alterar o `PenhasButtonStyle` com o novo estilo
2. Criar `extension` do `PenhasButtonStyle` para incluir o estilo que pertença a uma situação localizada

Qual é o sentido de migrar algo que foi implementado respeitando o SR para esta abordagem? Neste contexto considero apenas a opção **1** ou **3** a ser implementado.

Reply:
@shonorio não entendi o que faz que essa abordagem fira o SRP no caso do `PenhasButtonStyle` e já no `PenhasTextStyle` não?